src/DynamoCore/Graph/Nodes/CodeBlockNode.cs

```diff
@@ -1059,6 +1068,29 @@ private void LoadAndCreateConnectors(OrderedDictionary inportConnections, Ordere
                 undefinedIndices.RemoveAt(0);
                 unusedConnections.RemoveAt(0);
             }
+
+            List<List<ConnectorPinModel>> inportPinsList = inportPins.Values.Cast<List<ConnectorPinModel>>().ToList();
+            foreach (var list in inportPinsList)
+            {
+                foreach (var pin in list)
+                {
+                    var matchingConnector = this.AllConnectors.FirstOrDefault(c => c.GUID == pin.ConnectorId);
+                    if (matchingConnector is null) return;
+                    matchingConnector.AddPin(pin.CenterX, pin.CenterY);
+                }
+            }
+
+            List<List<ConnectorPinModel>> outportPinsList = outportPins.Values.Cast<List<ConnectorPinModel>>().ToList();
+            foreach (var list in outportPinsList)
+            {
+                foreach (var pin in list)
+                {
+                    var matchingConnector = this.AllConnectors.FirstOrDefault(c => c.GUID == pin.ConnectorId);
+                    if (matchingConnector is null) return;
+
+                    matchingConnector.AddPin(pin.CenterX, pin.CenterY);
+                }
+            }
```

Comment:
could we turn this into a private function so we dont have to duplicate it for the two collections?

Reply:
Done.